ctapipe/io/hdftableio.py

```diff
@@ -385,9 +428,18 @@ def __init__(self, filename):
         """
         super().__init__()
         self._tables = {}
-        self._h5file = tables.open_file(filename)
+        self._h5file = self.open(filename)
         pass
 
+    def open(self, filename):
+
+        self._h5file = tables.open_file(filename)
+        return self._h5file
```

Comment:
why the return?

Reply:
yes I think we can remove it

Reply:
hum actully I don't like much that class attributes are definded outside of the __init__()

so I will make open() return the table without assigning it to the class attribute

Reply:
I don't think this makes sense. 

I think it's perfectly alright for a method called `open` to set internal state of a class

Reply:
ok!